### PR TITLE
Added Kokkos compatibility documentation

### PR DESCRIPTION
**Pull request by nliber**
Initial draft of what we mean by Kokkos compatibility

This is to address issue #[4985](https://github.com/kokkos/kokkos/issues/4985)

@ldh4

### CODE REVIEW

**Comment by JBludau** [comment]
From slack:
Francesco: where is it written in the documentation/wiki that it is not legal to include a subdirectory?

Maybe we should mention that even though it is implicitly specified by saying "Kokkos_core.hpp" is the only public header. Might make the document more complete ...

**Comment by nliber** [comment]
I will say something along the lines of not to directly include things in subdirectories.

Do we want to encourage/discourage/say nothing about looking in other headers to see the actual public interface?  In practical terms, the headers are in some sense always up to date, but they may accidentally have public things we really don't want users to rely upon.

**Comment by nliber** [comment]
On a different library project, it was brought up that user defined macros could interfere with variable names, member functions, etc., including private ones.

I'll add something along the lines of user macros should also be prefaced with MYPROJECT_ (all cap unique name for the project followed by an underscore) to avoid collisions, either in the macro section or the very last section (which is essentially misc. rules).

**Comment by fnrizzi** [comment]
@nliber I took the liberty of opening #61 which also serves as an example for next documentation session

**Comment by PhilMiller** [comment]
There's a concern about how long compilation takes if we require all client code to include Kokkos_Core.hpp rather than exposing narrower headers as public. I recognize this isn't really the place to discuss that, though

File: Compatibility.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+## API Compatibility
+
+The public supported interface for Kokkos is:
+
+* The top-level `namespace Kokkos`
+* Macros starting with `KOKKOS_` (excluding those starting with `KOKKOS_IMPL_`)
+
+While the implementation details may change, the Kokkos Team puts its best effort into limiting changes to either having no functional behavioral differences (apart from bug fixes), or if they must, are changed with compile-time (preferred) or run-time warning and a suitable deprecation period (if possible).
+
+The experimentally support interface for Kokkos is in
```
[the remark this answers is not in the document]
Comment (nliber) [reply Compatibility.md:30]: Resolved in #73

**Review by masterleinad — CHANGES_REQUESTED** [review]
Use the new directories.

**Comment by nliber** [comment]
Superceded by PR #136.

**Pull request closed without merging**